File: php/bottle/Bottle.php
<?php

class Bottles
{
    public function song()
    {
        return $this->verses(99, 0);
    }

    private function verses(int $hi, int $lo)
    {
        $poem = '';
        for ( $n = $hi; $n >= $lo; $n--) {
            $poem .=$this->verse($n);
            $poem .= '</br>';
        }

        return $poem;
    }

    private function verse(int $n)
    {
        $verse = '';

        $verse .=  $n == 0 ? 'No more bottles' :  ( $n == 1) ? '1 bottle' :  $n . ' bottles';
        $verse .= ' of beer on the wall, ';
        $verse .=  $n == 0 ? 'No more bottles' :  ( $n == 1) ? '1 bottle' :  $n . ' bottles' . ' of beer.';
        $verse .= '</br>';
        $verse .=  $n == 1 ? 'Go to the store and buy some more, ' : 'Take one down and pass it around, ';
        $verse .=  $n == 1 ? 'No more bottles' :  ( $n == 2) ? '1 bottle' :  $n == 0 ? '99 bottles' : ( $n - 1) . ' bottles';
        $verse .= ' of beer on the wall.';
        $verse .= '</br>';

        return $verse;
    }
}

File: java/src/com/framgia/Part3/bottle/Bottle.java
package com.framgia.Part3.bottle;

public class Bottle {

    public static String song() {
        return getVerses(99, 0);
    }

    public static String getVerses(int start, int end) {
        StringBuilder poem = new StringBuilder();
        for (int i = start; i >= end; i--) {
            poem.append(getVerse(i));
            poem.append("\n");
        }
        return poem.toString();
    }

    public static String getVerse(int number) {
        switch (number) {
            case 0:
                return "No more bottles of beer on the wall, no more bottles of beer." + "\n" +
                        "Go to the store and buy some more, 99 bottles of beer on the wall.";
            case 1:
                return "1 bottle of beer on the wall, 1 bottle of beer." + "\n" +
                        "Take one down and pass it around, no more bottles of beer on the wall.";
            case 2:
                return "2 bottles of beer on the wall, 2 bottles of beer." + "\n" +
                        "Take one down and pass it around, 1 bottle of beer on the wall.";
            default:
                return number + " bottles of beer on the wall, " + number + " bottles of beer." + "\n" +
                        "Take one down and pass it around, " + (number - 1) + " bottles of beer on the wall.";
        }
    }

    public static void main(String[] args) {
        System.out.println(song());
    }
}